src/uucore/src/lib/features/format/num_format.rs

```diff
@@ -78,20 +78,36 @@ impl Formatter for SignedInt {
     type Input = i64;
 
     fn fmt(&self, mut writer: impl Write, x: Self::Input) -> std::io::Result<()> {
-        if x >= 0 {
-            match self.positive_sign {
-                PositiveSign::None => Ok(()),
-                PositiveSign::Plus => write!(writer, "+"),
-                PositiveSign::Space => write!(writer, " "),
-            }?;
-        }
+        let mut s = if self.precision > 0 {
+            format!("{x:0>width$}", width = self.precision)
+        } else {
+            x.to_string()
+        };
 
-        let s = format!("{:0width$}", x, width = self.precision);
+        let flag = if x >= 0 {
+            match self.positive_sign {
+                PositiveSign::None => String::new(),
+                PositiveSign::Plus => String::from("+"),
+                PositiveSign::Space => String::from(" "),
+            }
+        } else {
+            s = s[1..].to_string();
+            String::from("-")
+        };
 
+        let remaining_width = if self.width >= flag.len() {
+            self.width - flag.len()
+        } else {
+            0
+        };
```

Comment:
please add some comments explaining what it is doing

Reply:
Sure. Just added some comments.